tokio-boring/src/lib.rs

```diff
@@ -367,13 +385,18 @@ where
             stream: inner.stream,
             context: ctx as *mut _ as usize,
         };
+
         match (inner.f)(stream) {
             Ok(mut s) => {
                 s.get_mut().context = 0;
+                s.ssl_mut().set_ex_data(*TASK_CONTEXT_INDEX, 0);
```

Comment:
I do not like this. If there is a panic, neither of those will be reset to 0, which I think can lead to soundness issues afterwards, if you catch the unwind and try to use the boring API with the `SslContext` that uses async callbacks.

Reply:
Panics are segfaults in boring currently, added a note comment.